gcc/config/aarch64/aarch64-cores.def

```diff
@@ -170,12 +170,12 @@ AARCH64_CORE("cortex-r82", cortexr82, cortexa53, V8R, (), cortexa53, 0x41, 0xd15
    Apple implementer ID from xnu,
    guesses for part #, guesses for scheduler ident, generic_armv8_a for costs.
    A12 seems mostly 8.3,
-   M1 seems to be 8.4 + extras (see comments in option-extensions about f16fml),
-   M2 mostly 8.5 but with missing mandatory features.  */
+   M1 is based on 8.5 but lacks BTI, so we pretend it is 8.4,
+   M2 and M3 are based on 8.6.  */
 AARCH64_CORE("apple-a12", applea12, cortexa53, V8_3A,  (), generic_armv8_a, 0x61, 0x12, -1)
-AARCH64_CORE("apple-m1", applem1, cortexa57, V8_4A,  (F16, SB, SSBS), generic_armv8_a, 0x61, 0x23, -1)
-AARCH64_CORE("apple-m2", applem2, cortexa57, V8_4A,  (I8MM, BF16, F16, SB, SSBS), generic_armv8_a, 0x61, 0x23, -1)
-AARCH64_CORE("apple-m3", applem3, cortexa57, V8_4A,  (I8MM, BF16, F16, SB, SSBS), generic_armv8_a, 0x61, 0x23, -1)
+AARCH64_CORE("apple-m1", applem1, cortexa57, V8_4A,  (AES, DOTPROD, F16, RCPC, SB, SHA3, SSBS), generic_armv8_a, 0x61, 0x23, -1)
```

Comment:
See gcc/config/aarch64/aarch64-arches.def ...
you do not need to specify features already included in the arch (and that's also why we should not claim an arch that is incomplete)